### PR TITLE
bump aws lb controller to 2.2.0

### PR DESCRIPTION
**Pull request by olemarkus**
With these changes, we also pass the upstream e2e tests (with some amendments related to them expecting something more similar to EKS).

### CODE REVIEW

File: pkg/model/iam/iam_builder.go [hunk 1]
```diff
@@ -736,21 +736,33 @@ func addCalicoSrcDstCheckPermissions(p *Policy) {
...
+			Effect: StatementEffectAllow,
+			Action: stringorslice.Of(
+				"ec2:AuthorizeSecurityGroupIngress", // aws.go
+				"ec2:DeleteSecurityGroup",           // aws.go
+				"ec2:RevokeSecurityGroupIngress",    // aws.go
 
+				"elasticloadbalancing:ModifyTargetGroupAttributes",
+				"elasticloadbalancing:ModifyRule",
+				"elasticloadbalancing:DeleteRule",
+				"elasticloadbalancing:RemoveTags",
```
Comment (rifelpet) [inline pkg/model/iam/iam_builder.go:757]: `RemoveTags` is being added in both statements. Can we consolidate it to just one statement?

File: tests/e2e/scenarios/aws-lb-controller/run-test.sh [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+cd "$(mktemp -dt kops.XXXXXXXXX)"
+go get github.com/onsi/ginkgo/ginkgo
+
+git clone --branch e2e-filter-non-eligible-targets https://github.com/olemarkus/aws-load-balancer-controller .
+
+ginkgo -v -r test/e2e -- \
+    -cluster-name="${CLUSTER_NAME}" \
+    -aws-region="${REGION}" \
+    -aws-vpc-id="$VPC" \
+	-ginkgo.reportFile="${REPORT_DIR}/junit.xml"
```
Comment (rifelpet) [inline tests/e2e/scenarios/aws-lb-controller/run-test.sh:70]:
```suggestion
    -ginkgo.reportFile="${REPORT_DIR}/junit.xml"

```

File: tests/e2e/scenarios/aws-lb-controller/run-test.sh [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+VPC=$(${KOPS} toolbox dump -o json | jq -r .vpc.id)
+
+ZONE=$(${KOPS} get ig -o json | jq -r '[.[] | select(.spec.role=="Node") | .spec.subnets[0]][0]')
+
+REGION=${ZONE::-1}
+
+cd "$(mktemp -dt kops.XXXXXXXXX)"
+go get github.com/onsi/ginkgo/ginkgo
+
+git clone --branch e2e-filter-non-eligible-targets https://github.com/olemarkus/aws-load-balancer-controller .
```
Comment (rifelpet) [inline tests/e2e/scenarios/aws-lb-controller/run-test.sh:64]: Might be worth a comment linking to https://github.com/kubernetes-sigs/aws-load-balancer-controller/pull/2012, suggesting we can switch back to upstream/main when it has been merged.

File: pkg/model/iam/iam_builder.go (hunk 1 above)
Comment (olemarkus) [reply pkg/model/iam/iam_builder.go:757]: Hm. Is the last statement redundant? Technically they should be for different types of resources (ELBs vs listeners)

Comment (olemarkus) [reply pkg/model/iam/iam_builder.go:759]: Think this should work just as well.

**Review by rifelpet — APPROVED** [review]

**Pull request merged**